Code/Framework/AzManipulatorTestFramework/Include/AzManipulatorTestFramework/ActionDispatcher.h

```diff
@@ -79,89 +79,104 @@ namespace AzManipulatorTestFramework
+    template<typename DerivedDispatcherT>
+    void ActionDispatcher<DerivedDispatcherT>::Log(const char* format, ...)
     {
+        va_list args;
+        va_start(args, format);
+
         if (m_logging)
         {
-            AZ_Printf("ActionDispatcher", format, args...);
+            vprintf(format, args);
+            printf("%s", "\n");
         }
+
+        va_end(args);
     }
 
-    template <typename DerivedDispatcherT>
-    template <typename... Args>
-    DerivedDispatcherT* ActionDispatcher<DerivedDispatcherT>::Trace(const char* format, const Args&... args)
+    template<typename DerivedDispatcherT>
+    DerivedDispatcherT* ActionDispatcher<DerivedDispatcherT>::Trace(const char* format, ...)
     {
-        Log(format, args...);
+        va_list args;
+        va_start(args, format);
+
+        if (m_logging)
+        {
+            vprintf(format, args);
+            printf("%s", "\n");
+        }
+
+        va_end(args);
+
         return static_cast<DerivedDispatcherT*>(this);
     }
 
-    template <typename DerivedDispatcherT>
+    template<typename DerivedDispatcherT>
     DerivedDispatcherT* ActionDispatcher<DerivedDispatcherT>::EnableSnapToGrid()
     {
         Log("Enabling SnapToGrid");
         EnableSnapToGridImpl();
         return static_cast<DerivedDispatcherT*>(this);
     }
 
-    template <typename DerivedDispatcherT>
+    template<typename DerivedDispatcherT>
     DerivedDispatcherT* ActionDispatcher<DerivedDispatcherT>::DisableSnapToGrid()
     {
         Log("Disabling SnapToGrid");
         DisableSnapToGridImpl();
         return static_cast<DerivedDispatcherT*>(this);
     }
 
-    template <typename DerivedDispatcherT>
+    template<typename DerivedDispatcherT>
     DerivedDispatcherT* ActionDispatcher<DerivedDispatcherT>::GridSize(float size)
     {
         Log("GridSize: %f", size);
         GridSizeImpl(size);
         return static_cast<DerivedDispatcherT*>(this);
     }
 
-    template <typename DerivedDispatcherT>
+    template<typename DerivedDispatcherT>
     DerivedDispatcherT* ActionDispatcher<DerivedDispatcherT>::LogActions(bool logging)
     {
         m_logging = logging;
         Log("Log actions: %s", m_logging ? "enabled" : "disabled");
         return static_cast<DerivedDispatcherT*>(this);
     }
 
-    template <typename DerivedDispatcherT>
+    template<typename DerivedDispatcherT>
     DerivedDispatcherT* ActionDispatcher<DerivedDispatcherT>::CameraState(const AzFramework::CameraState& cameraState)
     {
-        Log("Camera state: p(%f, %f, %f) d(%f, %f, %f)",
-            float(cameraState.m_position.GetX()), float(cameraState.m_position.GetY()), float(cameraState.m_position.GetZ()),
-            float(cameraState.m_forward.GetX()), float(cameraState.m_forward.GetY()), float(cameraState.m_forward.GetZ()));
+        Log("Camera state: p(%f, %f, %f) d(%f, %f, %f)", float(cameraState.m_position.GetX()), float(cameraState.m_position.GetY()),
```

Comment:
Do you need the float casts here or was that from the days of VectorFloat?

Reply:
Oh that's a good point, these can likely be removed :+1:

Reply:
Now updated 👍